assets/src/edit-story/components/inspector/documentInspector.js

```diff
@@ -1,9 +1,187 @@
+		state: { users, statuses, disabledStatuses },
+	} = useInspector();
+
+	const {
+		state: { meta: { isSaving }, story: { author, status, slug, date, excerpt, featuredMediaUrl, password }, capabilities },
+		actions: { updateStory, deleteStory },
+	} = useStory();
+
+	const { postThumbnails } = useConfig();
+
+	useEffect( () => {
+		loadStatuses();
+		loadUsers();
+	} );
+
+	useEffect( () => {
+		const states = ( status === 'future' ) ? [ 'pending' ] : [ 'future', 'pending' ];
+		setDisabledStatuses( states );
+	}, [ status, setDisabledStatuses ] );
+
+	const handleChangeStatus = useCallback(
+		( evt ) => 	updateStory( { properties: { status: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeAuthor = useCallback(
+		( evt ) => updateStory( { properties: { author: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangePassword = useCallback(
+		( value, evt ) => updateStory( { properties: { password: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeDate = useCallback(
+		( value, evt ) => updateStory( { properties: { date: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeExcerpt = useCallback(
+		( value, evt ) => updateStory( { properties: { excerpt: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeSlug = useCallback(
+		( value, evt ) => updateStory( { properties: { slug: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeImage = useCallback(
+		( image ) => updateStory( { properties: { featuredMedia: image.id, featuredMediaUrl: ( image.sizes && image.sizes.medium ) ? image.sizes.medium.url : image.url } } ),
+		[ updateStory ],
+	);
+
+	const handleRemoveImage = useCallback(
+		( evt ) => {
+			updateStory( { properties: { featuredMedia: 0, featuredMediaUrl: '' } } );
+			evt.preventDefault();
+		},	[ updateStory ],
+	);
+
+	const handleRemoveStory = useCallback(
+		( evt ) => {
+			deleteStory();
+			evt.preventDefault();
+		},	[ deleteStory ],
+	);
+
 	return (
-		<h2>
-			{ 'Document' }
-		</h2>
+		<>
+			<h2>
+				{ 'Document' }
+			</h2>
+			{ capabilities && capabilities.hasPublishAction && statuses && <SelectMenu
+				label="Status"
+				name="status"
+				options={ statuses }
+				disabled={ isSaving }
+				disabledOptions={ disabledStatuses }
+				value={ status }
+				onChange={ handleChangeStatus }
+			/> }
+			{ capabilities && capabilities.hasPublishAction && status !== 'private' && <InputGroup
+				label={ 'Password' }
+				type={ 'password' }
+				value={ password }
+				disabled={ isSaving }
+				onChange={ handleChangePassword }
+			/> }
+
+			<RemoveButton onClick={ handleRemoveStory } dangerouslySetInnerHTML={ { __html: 'Move to trash' } } />
+			<InputGroup
+				label={ 'Published date' }
+				type={ 'datetime-local' }
+				value={ date }
+				disabled={ isSaving }
+				onChange={ handleChangeDate }
+			/>
+			{ capabilities && capabilities.hasAssignAuthorAction && users && <SelectMenu
```

Comment:
It seems like something weird is going on with the capability checks.

1. Edit story.
1. Assign story to user with subscriber role
1. Update story
1. Click Undo button.
1. See that the author field and others disappear unless the page is reloaded.

This might be an issue with Undo and not the capability checks added here, but I wanted to flag it either way.

Reply:
Ah, looks like @miina flagged the same thing too :-)

Resolving this as a duplicate.

Reply:
Fixed in https://github.com/ampproject/amp-wp/commit/62ac276fac3a40c6f2c60b7af27030ccb849d60c